extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -951,9 +946,14 @@ export const popupMessageListener = (request: Request) => {
     };
   };
 
-  const loadSettings = () => {
-    const dataSharingValue = localStorage.getItem(DATA_SHARING_ID) || "true";
-    const isDataSharingAllowed = JSON.parse(dataSharingValue);
+  const loadSettings = async () => {
+    await migrateLocalStorageToBrowserStorage();
+
+    const {
+      [DATA_SHARING_ID]: isDataSharingAllowed,
+    } = await dataStorage.getItem({
+      [DATA_SHARING_ID]: true,
+    });
```

Comment:
this is a test case for storing something in the new data store!

`isDataSharingAllowed` will be stored and fetched from dataStorage.